neurons/miner.py

```diff
@@ -159,7 +162,8 @@ def run(self):
                 self.metagraph_updater_thread.join()
                 self.position_inspector.stop_update_loop()
                 self.position_inspector_thread.join()
-                self.dashboard_thread.join()
+                if self.dashboard_thread is not None and self.dashboard_thread.is_alive():
```

Comment:
would it make sense to run some kind of exponential backoff here? When would we expect the thread to be dead

Reply:
The thread will not start if it's unable to find a free port. It dies if the dash happens to throw an exception. We could do an exponential backoff if we want it to continue trying the same ports again, but right now it's set up to try each port once before moving on.